src/java.base/share/classes/java/lang/Throwable.java

```diff
@@ -763,6 +767,45 @@ public void printStackTrace(PrintWriter s) {
         printStackTrace(new WrappedPrintWriter(s));
     }
 
+    /**
+     * Use a StringBuilder to print multiple StackTraceElements
+     */
+    private static void printStackTrace0(PrintStreamOrWriter printer, StackTraceElement[] trace) {
+        printStackTrace0(printer, null, trace, trace.length - 1, 0);
+    }
+
+    /**
+     * Use a StringBuilder to print multiple StackTraceElements
+     */
+    private static void printStackTrace0(
+            PrintStreamOrWriter printer,
+            String prefix,
+            StackTraceElement[] trace,
+            int traceEnd,
+            int framesInCommon) {
```

Comment:
You just need to pass `framesInCommon`, can calculate `int limit = trace.length - framesInCommon;` and then for the loop below, `for (int i = 0; i < limit; i++)` instead of using `traceEnd`

Reply:
According to your suggestion, I removed the passing of framesInCommon, which can be calculated by `trace.length - 1 - traceEnd`